File: textintel/visual/similarity.py
from __future__ import annotations

from textintel.lexical.character import combined_character_similarity
from textintel.visual.homoglyph import confusable_skeleton


def visual_similarity(a: str, b: str) -> float:
    sa = confusable_skeleton(a).casefold()
    sb = confusable_skeleton(b).casefold()
    return combined_character_similarity(sa, sb)

File: src/visual/similarity.rs
use crate::lexical::character::combined_character_similarity;
use crate::normalization::unicode::casefold_text;
use crate::visual::homoglyph::confusable_skeleton;

pub fn visual_similarity(a: &str, b: &str) -> f64 {
    let sa = casefold_text(&confusable_skeleton(a));
    let sb = casefold_text(&confusable_skeleton(b));
    combined_character_similarity(&sa, &sb)
}
